draft/ZEP0006.md

```diff
@@ -0,0 +1,248 @@
+        }
+    }
+  }
+}
+```
+
+The ZOM itself can also be represented as a JSON schema. Here is a the ZOM for Zarr V2 expressed as a JSON schema:
+```json
+{
+  "$ref": "#/definitions/Group",
+  "definitions": {
+    "Array": {
+      "title": "Array",
+      "description": "Model of a Zarr Version 2 Array",
+      "type": "object",
+      "properties": {
+        "attrs": {
+          "title": "Attrs",
+          "type": "object"
+        },
+        "shape": {
+          "title": "Shape",
+          "type": "array",
+          "items": {
+            "type": "integer"
+          }
+        },
+        "chunks": {
+          "title": "Chunks",
+          "type": "array",
+          "items": {
+            "type": "integer"
+          }
+        },
+        "dtype": {
+          "title": "Dtype",
+          "anyOf": [
+            {
+              "type": "string"
+            },
+            {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          ]
+        },
+        "compressor": {
+          "title": "Compressor",
+          "type": "object"
+        },
+        "fill_value": {
+          "title": "Fill Value"
+        },
+        "order": {
+          "title": "Order",
+          "enum": [
+            "C",
+            "F"
+          ],
+          "type": "string"
+        },
+        "filters": {
+          "title": "Filters",
+          "type": "array",
+          "items": {
+            "type": "object"
+          }
+        },
+        "dimension_separator": {
+          "title": "Dimension Separator",
+          "enum": [
+            ".",
+            "/"
+          ],
+          "type": "string"
+        },
+        "zarr_version": {
+          "title": "Zarr Version",
+          "default": 2,
+          "type": "integer"
+        }
+      },
+      "required": [
+        "attrs",
+        "shape",
+        "chunks",
+        "dtype",
+        "compressor",
+        "order",
+        "filters"
+      ],
+      "additionalProperties": false
+    },
+    "Group": {
+      "title": "Group",
+      "description": "Model of a Zarr Version 2 Group",
+      "type": "object",
+      "properties": {
+        "attrs": {
+          "title": "Attrs",
+          "type": "object"
+        },
+        "members": {
+          "title": "Members",
+          "type": "object",
+          "additionalProperties": {
+            "anyOf": [
+              {
+                "$ref": "#/definitions/Array"
+              },
+              {
+                "$ref": "#/definitions/Group"
+              }
+            ]
+          }
+        },
+        "zarr_version": {
+          "title": "Zarr Version",
+          "default": 2,
+          "type": "integer"
+        }
+      },
+      "required": [
+        "attrs",
+        "members"
+      ],
+      "additionalProperties": false
+    }
+  }
+}
+```
+
+And Zarr V3:
+
+```json
+# insert schema for v3 here
```

Comment:
let me know if you could use some help generating this.

Reply:
some help here would be great, thank you!